### ISSUE TITLE
Route-matching themes aren't respected on first load/refresh

### ISSUE THREAD

**Issue by ybnd** [DSpace/dspace-angular#1668]
**Describe the bug**
While applying #1627 to a client repository I noticed that it broke route-matching themes in a way I didn't picked up on before:
* `ThemeService.updateThemeOnRouteChange$` is no longer triggered when a page is first loaded, so the default theme gets used instead.
* Navigation is not affected

**To Reproduce**
1. Run a local DSpace instance with the following configuration:
   ```yaml
     themes:
      - name: custom
        regex: search
      - name: dspace
   ```
2. Open the app at `/home`
   * It should use the `dspace` theme
3. _Navigate_ to the `/search` page
   * It should use the `custom` theme
4. _Refresh_ the `/search` page
   * **It uses the `dspace` theme instead of the `custom` theme**

**Expected behavior**
Route-matching themes should always be used on the matching routes.

**Related work**
* #1627 
* #1326

**Comment by ybnd** [comment]
@tdonohue this is a pretty major problem with #1627 that I should have caught, I think it's crucial that we merge a fix before 7.3

**Comment by tdonohue** [comment]
@ybnd : Feel free to work on it, I'll pull it over onto the 7.3 board.  Bug fixes are always welcome, so if you can find a fix, please submit as soon as you can.
